src/vs/editor/contrib/stickyScroll/browser/stickyScrollController.ts

```diff
@@ -3,63 +3,281 @@
 		return this._widgetState;
 	}
 
+	// Get function to allow actions to use the sticky scroll controller
+	public static get(editor: ICodeEditor): IStickyScrollController | null {
+		return editor.getContribution<StickyScrollController>(StickyScrollController.ID);
+	}
+
+	private _disposeFocusStickyScrollStore() {
+		this._focused = false;
+		this._stickyScrollFocusedContextKey.set(false);
+		this._focusedStickyElement?.classList.remove('focus');
+		this._stickyScrollWidget.getDomNode().blur();
+		this._focusDisposableStore!.dispose();
+	}
+
+	public focus(): void {
+		this._focused = true;
+		const focusState = this._stickyScrollFocusedContextKey.get();
+		const rootNode = this._stickyScrollWidget.getDomNode();
+		this._stickyElements = rootNode.children;
+		this._numberStickyElements = this._stickyElements.length;
+
+		if (focusState === true || this._numberStickyElements === 0) {
+			// If lready focused or no line to focus on, return
+			return;
+		}
+		this._focusDisposableStore = new DisposableStore();
+		this._stickyScrollFocusedContextKey.set(true);
+		this._focusedStickyElement = rootNode.lastElementChild! as HTMLDivElement;
+		this._focusedStickyElement.classList.add('focus');
+		// The first line focused is the bottom most line
+		this._focusedStickyElementIndex = this._numberStickyElements - 1;
+
+		// Whenever the mouse hovers on the sticky scroll remove the keyboard focus
+		this._focusDisposableStore.add(dom.addDisposableListener(rootNode, dom.EventType.MOUSE_OVER, () => {
+			this._focusedStickyElement?.classList.remove('focus');
+		}));
+		// Whenever the mouse leaves the sticky scroll, set the keyboard focus to be on the last line hovered
+		this._focusDisposableStore.add(dom.addDisposableListener(rootNode, dom.EventType.MOUSE_OUT, () => {
+			this._focusedStickyElement = this._stickyScrollWidget.lastMouseFocusedStickyLine;
+			this._focusedStickyElementIndex = this._stickyScrollWidget.lastMouseFocusedStickyLineIndex;
+			this._focusedStickyElement?.classList.add('focus');
+		}));
+
+		// When scrolling, remove the focus
+		this._focusDisposableStore.add(this._editor.onDidScrollChange(() => {
+			this._disposeFocusStickyScrollStore();
+		}));
+		rootNode.focus();
+	}
+
+	public focusNext(): void {
+		if (this._focusedStickyElement && this._focusedStickyElementIndex! < this._numberStickyElements! - 1) {
+			this._focusNav(true);
+		}
+	}
+
+	public focusPrevious(): void {
+		if (this._focusedStickyElement && this._focusedStickyElementIndex! > 0) {
+			this._focusNav(false);
+		}
+	}
+
+	// True is next, false is previous
+	private _focusNav(direction: boolean): void {
+		this._focusedStickyElement!.classList.remove('focus');
+		this._focusedStickyElementIndex = direction ? this._focusedStickyElementIndex! + 1 : this._focusedStickyElementIndex! - 1;
+		this._focusedStickyElement = this._stickyElements!.item(this._focusedStickyElementIndex!)! as HTMLDivElement;
+		this._focusedStickyElement.classList.add('focus');
```

Comment:
also call `this._focusedStickyElement.focus()` to ensure the child element is focused